frontend/src/job/ingest-journal/ZenonImport.ts

```diff
@@ -0,0 +1,64 @@
+import axios from 'axios';
+
+const zenonBaseURL: string = 'https://zenon.dainst.org/api/v1/';
+
+function isId(term: string) {
+    return !!term.match(/^\w?\w?\d{9}$/);
```

Comment:
Leider bestehen aktuell nicht alle Zenon IDs aus 9 Ziffern. Es gibt immer noch einige Altdaten, die mit Buchstaben beginnen.

Reply:
Für beide deine Anmerkungen gilt, dass ich das aus dem alten Code von Philipp übernommen habe. Was wäre ier die richtige regex, einfach ein Buchstabe am anfang zulassen?

Reply:
Müsste ich Sabine fragen welche Varianten es da gibt. 

Allgemein frage ich mich aber, ob man das überhaupt braucht. Wenn man z.B. eine getRecord(id: string) Funktion einführen würde, dann macht man einfach den Versuch und falls ein 404 kommt, muss man den halt behandeln. Ich würde gar nicht beim Cilantro anfangen, irgendwelche Annahmen zu den Zenon-Daten zu machen.

Reply:
Aber woher weiss man denn ob man record oder seaarch machen will? Soll das der User auswählen?

Reply:
Ok, also soweit ich das sehe, hat Philipp das etwas komisch implementiert. 

Jetzt ein wenig Mutmaßung:
1. Philipp wollte dass man entweder nach dem Titel suchen oder eine Zenon ID eingeben kann.
2. Du kannst wenn du "Alle Felder" suchst, auch als Such-String eine Zenon ID eingeben. Da er aber explizit nach Titeln sucht, hat er den Regex mit anschließender Umleitung auf den Record Endpoint eingebaut.

Konzeptionell sauberer fände ich: 

- Ein Eingabefeld, wo man eine Zenon-ID eingeben kann, die dann auf die /record API geht. Dieses Eingabefeld kann man benutzen, wenn man **weiß** welches Record man haben will. Du hast ja quasi schon ein Feld dafür, vielleicht könnte man da im Hintergrund über die record-API überprüfen, ob es den Datensatz wirklich gibt, wenn da jemand was von Hand einträgt. Dafür wäre dann eine getRecord(id: string) Funktion zuständig.
- Ein echtes Suchfeld mit den 3 Optionen (s.u.), die dann auf den search Endpoint geht, wenn man eben noch **nicht weiß**, welchen Zenon-Datensatz man möchte.

Oder wir machen es erstmal einfach und implementieren nur das Suchfeld mit den 3 Optionen. Wie gesagt sollte eine Suche in allen Feldern auch funktionieren, wenn man als Suchbegriff die Zenon-ID sucht.